packages/api/src/services/license/mappers/license-response-mapper.ts

```diff
@@ -18,23 +17,27 @@ export class LicenseResponseMapper extends Mapper<License> {
    * @returns {License} - A new License instance
    */
   mapToModel(response: LicenseResponse): License {
-    const license = new License();
+    const licenseCapabilities = new CapabilityListMapper().mapToModel(
+      (response.licenseCapabilities ?? []).map(c => c as Capability)
+    );
 
-    license.expiryDate = response?.expiryDate ? new Date(response.expiryDate) : undefined;
-    license.latestPublicationDate = response?.latestPublicationDate ? new Date(response.latestPublicationDate) : undefined;
-    license.licensee = response?.licensee || '';
-    license.maxCpuCores = response?.maxCpuCores;
-    license.product = toProduct(response?.product);
-    license.productType = toProductType(response?.productType);
-    license.licenseCapabilities = MapperProvider.get(CapabilityListMapper).mapToModel(response?.licenseCapabilities);
-    license.version = response?.version || '';
-    license.installationId = response?.installationId || '';
-    license.valid = response?.valid;
-    license.typeOfUse = response?.typeOfUse || '';
-    license.message = response?.message || '';
-    license.present = response?.present || false;
-    license.usageRestriction = response?.usageRestriction || '';
-
-    return license;
+    return new License({
```

Comment:
This reminds me why I don't like the whole partial object as constructor argument idea.